lib/posix-user/user.c

```diff
@@ -228,13 +228,12 @@ int getresgid(gid_t *rgid, gid_t *egid, gid_t *sgid)
 	return 0;
 }
 
-int setresgid(gid_t rgid, gid_t egid, gid_t sgid)
+UK_SYSCALL_R_DEFINE(int, setresgid, gid_t, rgid, gid_t, egid, gid_t, sgid)
 {
 	/* We allow only UK_DEFAULT_GID */
 	if (rgid != UK_DEFAULT_GID || egid != UK_DEFAULT_GID ||
```

Comment:
Not really related to the purpose of this PR, but according to [man pages](https://man7.org/linux/man-pages/man2/setresuid.2.html): `If one of the arguments equals -1, the corresponding value is not changed.`.

In this case, shouldn't we do something like this?

```
if ((rgid != UK_DEFAULT_GID && rgid != -1) || (egid != UK_DEFAULT_GID && egid != -1) ||
		(sgid != UK_DEFAULT_GID && sgid != -1)) {
```